cmd/cloudshell_open/jib.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func jibMavenBuild(dir string, image string) error {
+	cmd := createMavenCommand(dir, "--batch-mode", "-Dmaven.test.skip=true",
+		"package", "jib:build", "-Dimage="+image, "-Djib.to.auth.credHelper=gcloud")
+	if b, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("Jib Maven build failed: %v, output:\n%s", err, string(b))
+	}
+	return nil
+}
+
+func jibMavenConfigured(dir string) (bool, error) {
+	if _, err := os.Stat(filepath.Join(dir, "pom.xml")); err != nil {
```

Comment:
if you actually unfold ioutil.ReadAll here, you can do os.IsNotExist() directly on its err, and therefore eliminate the os.Stat

Reply:
Cool. I've simplified the code around this.